includes/polling/ports/procera.inc.php

```diff
@@ -46,19 +46,33 @@
     'ifOutErrors' => 'TxErrors',
 );
 
+// Media Types as per PACKETLOGIC-CHANNEL-MIB
+$mediaType = array(
+    array('ifDuplex' => null,'ifSpeed' => 0, 'label'=> 'linkdown'),
+    array('ifDuplex' => 'halfDuplex', 'ifSpeed' => '10000000', 'label' => 'hd10'),
+    array('ifDuplex' => 'fullDuplex', 'ifSpeed' => '10000000', 'label' => 'fd10'),
+    array('ifDuplex' => 'halfDuplex', 'ifSpeed' => '100000000', 'label' => 'hd100'),
+    array('ifDuplex' => 'fullDuplex', 'ifSpeed' => '100000000', 'label' => 'fd100'),
+    array('ifDuplex' => 'fullDuplex', 'ifSpeed' => '1000000000', 'label' => 'fd1000'),
+    array('ifDuplex' => 'fullDuplex', 'ifSpeed' => '10000000000', 'label' => 'fd10000')
+);
+
+
 foreach ($packetlogic_stats as $index => $port) {
     $procera_port = array();
     foreach ($channelTypes as $cType) {
         foreach ($required as $ifEntry => $IfxStat) {
-            $procera_port[$ifEntry] = $packetlogic_stats[$index][0][$cType['type'].$IfxStat];
+            $procera_port[$ifEntry] = $packetlogic_stats[$index][$cType['type'].$IfxStat];
         }
-        $procera_port['ifName'] = $packetlogic_stats[$index][0]['channelName']. ' '.$cType['name'];
-        $procera_port['ifDescr'] = $packetlogic_stats[$index][0]['channelName']. ' '.$cType['name'];
-        $procera_port['ifConnectorPresent'] = ($packetlogic_stats[$index][0]['NegotiatedMedia'] != 'linkdown' ? "true" : "false");
-        $procera_port['ifOperStatus'] = ($packetlogic_stats[$index][0]['channelActive'] === 'active' ? "up" : "down");
+        $procera_port['ifName'] = $packetlogic_stats[$index]['channelName']. ' '.$cType['name'];
+        $procera_port['ifDescr'] = $packetlogic_stats[$index]['channelName']. ' '.$cType['name'];
+        $procera_port['ifConnectorPresent'] = ($packetlogic_stats[$index][$cType['type']."NegotiatedMedia"] != '0' ? "true" : "false");
+        $procera_port['ifOperStatus'] = ($packetlogic_stats[$index]['channelActive'] == true ? "up" : "down");
```

Comment:
`$packetlogic_stats[$index]['channelActive'] == 1` might be a little more clear.  This oid returns 1 or 0.